pkg/integrations/v4/supervisor_fb.go

```diff
@@ -144,6 +163,63 @@ func saveToTempFile(config []byte) (string, error) {
 	return file.Name(), nil
 }
 
+func removeFbConfigTempFiles(maxNumberOfFbConfigTempFiles int) ([]string, error) {
+
+	files, err := os.ReadDir(os.TempDir())
+	if err != nil {
+		return nil, err
+	}
+
+	var fbConfigTempFiles []fs.DirEntry
+
+	for _, file := range files {
+		if !file.IsDir() && strings.HasPrefix(file.Name(), "nr_fb_config") {
+			fbConfigTempFiles = append(fbConfigTempFiles, file)
+		}
+	}
+
+	if len(fbConfigTempFiles) > maxNumberOfFbConfigTempFiles {
+
+		var removedConfigTempFilenames []string
+
+		luaFilterTempFileRegex := regexp.MustCompile("nr_fb_lua_filter\\d+")
+
+		sort.Slice(fbConfigTempFiles, func(i, j int) bool {
+			fileInfo1, _ := fbConfigTempFiles[i].Info()
+			fileInfo2, _ := fbConfigTempFiles[j].Info()
+			return fileInfo1.ModTime().Before(fileInfo2.ModTime())
+		})
+
+		for i := 0; i < len(fbConfigTempFiles)-maxNumberOfFbConfigTempFiles; i++ {
```

Comment:
[Totally optional suggestion] I see you're iterating through the config files and at the same time removing all the LUA filters that each one of it contains. First you remove all the LUA files, and then the config file itself. So, you need to call the `os.Remove` twice, and deal with the potential errors twice. As a totally optional suggestion, you could refactor the method so that it first focuses on assembling an array of files to remove by just iterating the older config files and gathering the potential lua files, and then just iterate this array of files to remove and remove them.

Reply:
I agree with the proposed approach.

Also, I would not break the loop execution on error, as if in a weird scenario a file cannot be deleted, it will break the execution and no other file will be deleted.

I would capture all errors, and return all as they are logged later.